[File Explorer Add-ons] Fix file preview pane flickering on file selection and resizing

tanchekwei opened the pull request:
<!-- Enter a brief description/summary of your PR here. What does it fix/what does it change/how was it tested (even manually, if necessary)? -->
## Summary of the Pull Request

This PR fix file preview pane flickering issues in the File Explorer Addon for MonacoPreview, SvgPreview and MarkdownPreview. These previews utilize WebView2, which has been causing the flickering issues. The problems occur during file selection and resizing the file preview pane.

To address these issues, the PR applies [a fix from the Peek module](https://github.com/microsoft/PowerToys/pull/26342/files), following the suggestion of @crutkas. This fix flickering after the file preview loading screen is removed and before the file content is displayed (refer to the blue flickering [in the video](https://github.com/microsoft/PowerToys/assets/26683979/11df11b4-affb-49a1-8a31-396c0d150588)). However, it does not fix the flickering that when the user changes file selection or resizes the file preview pane. Because the flickering is not related to the WebView2 background color, as it occurs before the WebView is created.

To resolve the remaining issues, two modifications were made. Firstly, the background color was set for the parent window (fix green flickering [in the video](https://github.com/microsoft/PowerToys/assets/26683979/11df11b4-affb-49a1-8a31-396c0d150588)). Secondly, the background color was set in the form, after invoking the handler executable (fix red flickering [in the video](https://github.com/microsoft/PowerToys/assets/26683979/11df11b4-affb-49a1-8a31-396c0d150588)).

<!-- Please review the items on the PR checklist before submitting-->
## PR Checklist

- [x]  **Closes:** #20436
- [ ] **Communication:** I've discussed this with core contributors already. If work hasn't been agreed, this work might be rejected
- [x] **Tests:** ~~Added/updated and~~ all pass
- [ ] **Localization:** All end user facing strings can be localized
- [ ] **Dev docs:** Added/updated
- [ ] **New binaries:** Added on the required places
   - [ ] [JSON for signing](https://github.com/microsoft/PowerToys/blob/main/.pipelines/ESRPSigning_core.json) for new binaries
   - [ ] [WXS for installer](https://github.com/microsoft/PowerToys/blob/main/installer/PowerToysSetup/Product.wxs) for new binaries and localization folder
   - [ ] [YML for CI pipeline](https://github.com/microsoft/PowerToys/blob/main/.pipelines/ci/templates/build-powertoys-steps.yml) for new test projects
   - [ ] [YML for signed pipeline](https://github.com/microsoft/PowerToys/blob/main/.pipelines/release.yml)
- [ ] **Documentation updated:** If checked, please file a pull request on [our docs repo](https://github.com/MicrosoftDocs/windows-uwp/tree/docs/hub/powertoys) and link it here: #xxx

<!-- Provide a more detailed description of the PR, other things fixed or any additional comments/features here -->
## Detailed Description of the Pull Request / Additional comments

Other things fixed
- Use Color.FromArgb instead of ColorTranslator.FromHtml for improved performance ([dotnetfiddle](https://dotnetfiddle.net/k2CoiR)), put hex color codes in comment. https://github.com/microsoft/PowerToys/pull/26660/commits/e89a4f6c12a9119debe7c4c2cab41b9b9fd0e654

<!-- Describe how you validated the behavior. Add automated tests wherever possible, but list manual validation steps taken as well -->
## Validation Steps Performed

For the modules I changed, I done following steps for validation
1. Built the .MSI installer and installed it
2. Manual testing, I observe if my changes fix flickering on file selection and while resizing file preview pane.
3. File selection test: I select multiple files repeatedly and select the files then select folder repeatedly
4. File preview pane resizing test: I resize the pane slowly and rapidly to check for any flickering
5. Tested theme switching by switching between Light and Dark modes, as well as transitioning from Dark to Light and vice versa. To ensured the styles adjusted correctly based on the selected theme.
6. Tested my changes on different machines running Windows 11 and Windows 10.

Before

https://github.com/microsoft/PowerToys/assets/26683979/da8d1e31-6186-476c-8974-3d27c380236f

After (dark theme)

https://github.com/microsoft/PowerToys/assets/26683979/6b5e5d94-212b-447f-b3ac-18c9702ceaff

After (dark theme to light theme)

https://github.com/microsoft/PowerToys/assets/26683979/71fcb2b1-09c0-42a1-ab19-65c33e3ca9b6

tanchekwei committed:
Fix MonacoPreview flickering on file selection and resizing

tanchekwei commented:
@microsoft-github-policy-service agree

tanchekwei committed:
Fix MarkdownPreview flickering on file selection and resizing

tanchekwei committed:
Fix SvgPreview flickering on file selection and resizing

tanchekwei committed:
Create Settings class and standardize background setting for MarkdownPreview

tanchekwei committed:
Replace ColorTranslator.FromHtml with Color.FromArgb for constant color settings

tanchekwei committed:
Use existing SetBackground

tanchekwei committed:
Move color values to constant

jaimecbernardo reviewed (approved):
LGTM! Thank you for the contribution!
Just a nit, why add a "Settings.cs" to Markdown Preview Handler?

tanchekwei committed:
Remove duplicate GetTheme function

tanchekwei commented:
> LGTM! Thank you for the contribution! Just a nit, why add a "Settings.cs" to Markdown Preview Handler?

Thanks for the review! I added the Settings.cs to the Markdown Preview Handler because I noticed that the Monaco and SVG Preview Handler also have their own Settings class. This allows them to have a consistent way of getting the background color and theme.

I realized I had defined the GetTheme function in both the Settings and MarkdownPreviewHandler class. Sorry for that. I've fixed the issue and tested the changes.

Do the changes look good to you? Let me know if there's anything else I can do.

jaimecbernardo commented on src/modules/previewpane/MarkdownPreviewHandler/Settings.cs:
```diff
@@ -0,0 +1,42 @@
+    internal sealed class Settings
+    {
+        /// <summary>
+        /// Gets the color of the window background.
```
```suggestion
        /// Gets the color of the window background.
        /// Even though this is not a setting yet, it's retrieved from a "Settings" class to be aligned with other preview handlers that contain this setting.
        /// It's possible it can be converted into a setting in the future.
```

jaimecbernardo committed:
Update src/modules/previewpane/MarkdownPreviewHandler/Settings.cs

jaimecbernardo commented:
Thanks for the explanation. I've added a comment to explain why it's a Settings.cs file, just for clarity. After the PR passes CI I'll merge it. Thank you for the contribution!

stefansjfw commented:
@tanchekwei we had to revert your fix for this issue as it introduces some regressions within the new release we are preparing at the moment. We'll investigate it and see how to fix the regression for the release after.
